src/components/Table/TableFormatters/DrawerDetailFormatter.vue

```diff
@@ -0,0 +1,293 @@
+        return route.components.default
+      }
+    },
+    showDrawer() {
+      if (this.formatterArgs.drawerComponent) {
+        this.drawerComponent = this.formatterArgs.drawerComponent
+      } else {
+        this.drawerComponent = this.getRouteComponent()
+      }
+      const route = this.getDetailRoute()
+      if (route?.query?.tab) {
+        this.$cookie.set(route.name, route.query.tab, 1)
+        this.$route.query.tab = route.query.tab
+      }
+      const payload = {
+        action: 'detail',
+        row: this.row,
+        col: this.col,
+        id: route.params.id || this.row.id
+      }
+      this.$store.dispatch('common/setDrawerActionMeta', payload).then(() => {
+        this.drawerTitle = this.getDrawerTitle(payload)
+        this.ddrawerVisible = true
+      })
+    },
+    handleClick() {
+      if (this.formatterArgs.beforeClick) {
+        this.formatterArgs.beforeClick(this.callbackArgs)
+      }
+
+      if (this.formatterArgs.onClick) {
+        return this.formatterArgs.onClick(this.callbackArgs)
+      }
+
+      if (this.formatterArgs.drawer) {
+        return this.showDrawer()
+      }
+
+      if (this.preventClick) {
+        return
+      }
+
+      this.goDetail()
+    },
+    getDetailRoute() {
+      // const defaultRoute = this.$route.name.replace('List', 'Detail')
+      let route = this.formatterArgs.route
+      if (this.formatterArgs.getRoute && typeof this.formatterArgs.getRoute === 'function') {
+        route = this.formatterArgs.getRoute(this.callbackArgs)
+      }
+      if (!route) {
+        console.error('No route found')
+        return
+      }
+
+      let detailRoute = { replace: true }
+      if (typeof route === 'string') {
+        detailRoute.name = route
+        detailRoute.params = { id: this.row.id }
+      } else {
+        detailRoute = route
+      }
+
+      const routeQuery = this.formatterArgs.routeQuery
+      if (routeQuery && typeof routeQuery === 'object') {
+        detailRoute.query = this.formatterArgs.routeQuery
+      }
+      return detailRoute
+    },
+    goDetail() {
+      const detailRoute = this.getDetailRoute()
+
+      if (this.formatterArgs.openInNewPage) {
+        const { href } = this.$router.resolve(detailRoute)
+        this.linkClicked = this.formatterArgs.removeColorOnClick
+        return window.open(href, '_blank')
+      }
+
+      this.$router.push(detailRoute)
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.detail {
+  display: inline-block;
+  max-width: 100%;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  font-weight: 400;
+}
+
+.detail {
+  line-height: 25px;
+  font-size: 13px;
+}
+
+.clicked,
+.el-link.el-link--info.clicked {
+  color: inherit !important;
+}
+
+.icon {
+  width: 28px;
+  height: 28px;
+}
+
+::v-deep .go-back {
+  display: none;
+}
+
+.detail-drawer {
+  ::v-deep {
+    .el-drawer__header {
+      border-bottom: none;
+      padding-bottom: 1px;
+    }
+  }
+}
+
+</style>
```

Comment:
This Code is already clean and free of common mistakes like syntax errors or logic defects. There aren't any key areas where I would apply additional optimizations.